pygit2/repository.py

```diff
@@ -915,3 +904,22 @@ def set_ident(self, name, email):
 
         err = C.git_repository_set_ident(self._repo, to_bytes(name), to_bytes(email))
         check_error(err)
+
+
+class Repository(BaseRepository):
+    def __init__(self, path, *args, **kwargs):
+        if not isinstance(path, six.string_types):
+            path = path.decode('utf-8')
+
+        path_backend = init_file_backend(path)
+        super(Repository, self).__init__(backend=path_backend, *args, **kwargs)
+        self._common_init()
+
```

Comment:
``_common_init`` would be called twice, first from the super class, and then here